plantuml-parser-core/src/test/java/com/shuzijun/plantumlparser/core/TestRecordAndDefaultConstructor.java

```diff
@@ -0,0 +1,136 @@
+package com.shuzijun.plantumlparser.core;
+
+import com.github.javaparser.ParserConfiguration;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Scanner;
+
+public class TestRecordAndDefaultConstructor {
+
+    private static ParserConfig makeParserConfig(String filename, boolean showDefaultCon) {
+        ParserConfig parserConfig = new ParserConfig();
+        parserConfig.setLanguageLevel(ParserConfiguration.LanguageLevel.JAVA_17);
+        parserConfig.setShowConstructors(true);
+        parserConfig.setShowDefaultConstructors(showDefaultCon);
+        parserConfig.addFieldModifier("private");
+        parserConfig.addMethodModifier("public");
+        parserConfig.addMethodModifier("default");
+        File file = new File(Objects.requireNonNull(
+            TestDefaultConstructor.class.getClassLoader().getResource(filename)).getFile());
+        parserConfig.addFilePath(file.getAbsolutePath());
+        return parserConfig;
+    }
+
+    private static String getUml(String filename, boolean showDefaultCon) throws IOException {
+        ParserProgram parserProgram = new ParserProgram(makeParserConfig(filename, showDefaultCon));
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream oldOut = System.out;
+        System.setOut(new PrintStream(baos));
+        parserProgram.execute();
+        String result = baos.toString();
+        System.setOut(oldOut);
+        return result;
+    }
+
+    private static boolean isUmlEquivalent(String uml1, String uml2, String umlFilename)
+        throws FileNotFoundException {
+
+        if (uml2 != null && umlFilename != null) {
+            throw new IllegalArgumentException("Only one of these inputs is accepted.");
+        }
+
+        String[] lines1 = uml1.split("\n");
+
+        if (uml2 != null) {
+            // Test record
+            return isUmlEquivalentForRecord(lines1, uml2);
```

Comment:
I don't think there should be different `isUmlEquivalent` methods for records vs. default constructors. There's no reason for them to be different, and that implies that other language features would need their own methods. I think you should probably have:

```
boolean isUmlEquivalent(String uml1, String uml2)
```
and
```
boolean isUmlEquivalent(String uml1, String filename)
```
The second method would call the first one.